### DISCUSSION TITLE
Can't show Git history on OpenGrok on windows

### DISCUSSION THREAD

**Discussion by FridmanElad** [oracle/opengrok#4339] [General]
Hi

I have OpenGrok up and running with these:
Tomcat 9
Java 11
opengrok-1.5.11
ctags-p6.0.20230416.0-x64
Git
Git Repository = C:\OpenGrok_Sources\setup\\.git

My problem is that I want OpenGrok will support History of git but it doesn't work. The History ( and also Annotate ) are just text and not link and when I choose file and click on the H link I get error.

This is the index command I'm running:
"C:\Program Files\Zulu\zulu-11\bin\java" -Xmx1048m -jar C:\apps\opengrok-1.5.11\lib/opengrok.jar  -P -W C:\OpenGrok\run_update_SRC.xml -v -H -S -G --depth 8 -c C:\apps\ctags-p6.0.20230416.0-x64\ctags.exe -s "C:\OpenGrok_Sources\setup" -d "C:\opengrok\grokdata\SRC"

Any idea what I'm doing wrong ?

**Comment by vladak** [comment]
The first thing I noticed is that if the source root directory is `C:\OpenGrok_Sources\setup` then `C:\OpenGrok_Sources\setup.git` does not look like a sub-directory of that.

Is there anything of interest in the webapp logs when the xref is displayed ? If not, try raising the log level per https://github.com/oracle/opengrok/wiki/Debugging#web-application and retry the request and see what was added to the Tomcat log.

Does this happen with the latest OpenGrok version ?

**Reply by vladak** [reply]
Also:
>  when I choose file and click on the H link I get error.

what sort of error exactly ?

**Comment by FridmanElad** [comment]
1. That was a typo
Under C:\OpenGrok_Sources\setup I have this:
.git
COMMON
SETUP
.gitignore

And the sourceRoot is set to C:\OpenGrok_Sources\setup

2. Didn't see anything special in the log
3. I didn't managed to run it with opengrok-1.12.5, maybe need to upgrade to Tomcat 10 ?
4. That's the error I get:
Error: File not found!
The requested resource is not available.

Resource lacks history info. Was remote SCM side up when indexing occurred? Cleanup history cache dir(or just the .gz for the file or db record) and rerun indexer making sure remote side will respond during indexing.

Thanks

**Reply by vladak** [reply]
You need to raise log level for Tomcat to `FINEST` to see any relevant diagnostic messages. Yes, newer OpenGrok versions have Tomcat 10 as minimum version.

**Reply by vladak** [reply]
> 1. That was a typo
>        Under C:\OpenGrok_Sources\setup I have this:
>        .git
>        COMMON
>        SETUP
>        .gitignore

Given that `C:\OpenGrok_Sources\setup` is source root (set by the -s option of the indexer), the Git repository (signified by the `.git` directory) will not be detected. It needs to be in a sub-directory of the source root.

**Comment by FridmanElad** [comment]
I tried latest opengrok-1.12.5 with Tomcat 10 but get same result. I have a feeling that something is wrong with the .xml configuration file or batch file that create the index.
From where can I take examples that support Git ?

**Reply by vladak** [reply]
You'd need to raise the log level of the webapp to see any relevant log messages. Anyhow, see above for why the Git repository was not detected.

In fact, if you look into the XML configuration file generated by the indexer, normally there should be a `RepositoryInfo` object matching the path of the repository. I bet that in your case it is not there because of the reason explained above.

**Comment by FridmanElad** [comment]
I have moved the .git folder to be under C:\OpenGrok_Sources\setup\COMMON 
1. Now I see the history but it doesn't work as I expected,
For example I can see list of commits on the repo but only in the root folder, when I browse to some folder and click History the list is empty
2. I don't understand what should be the folder structure in case I have a git repo with multiple folders and each one of them should be a project under OpenGrok.

**Reply by vladak** [reply]
1. is the history empty for all files or just for some ? It might be worthwhile to reindex from scratch (i.e. delete the data root and reindex).
2. with projects enabled (-P), just checkout the repository into a directory underneath source root, i.e. something like this:
```
cd C:\OpenGrok_Sources\setup
git clone http://example.com foo
```
